addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -225,42 +227,42 @@ func (r *MirrorPeerReconciler) createVolumeReplicationClass(ctx context.Context,
 		klog.Error(err, "Failed to get current storage cluster ref")
 		return err
 	}
-	params := make(map[string]string)
-	params[MirroringModeKey] = string(mp.Spec.Mode)
-	params[SchedulingIntervalKey] = mp.Spec.SchedulingInterval
-	params[ReplicationSecretNameKey] = mp.Spec.ReplicationSecretName
-	params[ReplicationSecretNamespaceKey] = scr.Namespace
-	vrc := &replicationv1alpha1.VolumeReplicationClass{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "odf-rbd-volumereplicationclass",
-		},
-		Spec: replicationv1alpha1.VolumeReplicationClassSpec{
+
+	for _, interval := range mp.Spec.SchedulingInterval {
+		params := make(map[string]string)
+		params[MirroringModeKey] = DefaultMirroringMode
+		params[SchedulingIntervalKey] = interval
+		params[ReplicationSecretNameKey] = ReplicationSecretName
+		params[ReplicationSecretNamespaceKey] = scr.Namespace
+		vrcName := fmt.Sprintf(VolumeReplicationClassNameTemplate, common.FnvHash(interval))
+		found := &replicationv1alpha1.VolumeReplicationClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: vrcName,
+			},
+		}
+		err = r.SpokeClient.Get(ctx, types.NamespacedName{
+			Name: found.Name,
+		}, found)
+
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+		spec := replicationv1alpha1.VolumeReplicationClassSpec{
 			Provisioner: fmt.Sprintf(ProvisionerTemplate, scr.Namespace),
 			Parameters:  params,
-		},
-	}
-	found := &replicationv1alpha1.VolumeReplicationClass{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "odf-rbd-volumereplicationclass",
-		},
+		}
+		op, err := controllerutil.CreateOrUpdate(ctx, r.SpokeClient, found, func() error {
```

Comment:
update to specific parameters might not be supported in the future see https://github.com/csi-addons/volume-replication-operator/issues/88

Reply:
If we do create only, how do we handle upgrades where VRC spec might change?
We might need to label VRCs so that we can get list of VRC that we created and update it.

Reply:
You need to delete and recreate. VRC is the same as storageclass

Reply:
Good info to know. We will just create and forget then.